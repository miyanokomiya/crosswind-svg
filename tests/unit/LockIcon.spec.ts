import { mount } from "@vue/test-utils";
import Target from "../../src/components/LockIcon.vue";

describe("LockIcon", () => {
  it("locked:false", () => {
    const wrapper = mount(Target, {
      propsData: { locked: false }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("locked:true", () => {
    const wrapper = mount(Target, {
      propsData: { locked: true }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
