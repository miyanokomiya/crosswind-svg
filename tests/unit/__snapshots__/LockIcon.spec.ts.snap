// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LockIcon locked:false 1`] = `
<svg
  style="transform: rotate(0); fill: currentColor; stroke: currentColor; user-select: none;"
  viewBox="0 0 64 64"
  xmlns="http://www.w3.org/2000/svg"
  xmlns:xlink="http://www.w3.org/1999/xlink"
>
  <g
    fill="white"
  >
    <circle
      cx="32"
      cy="32"
      r="32"
    />
  </g>
   
  <g
    stroke-linecap="round"
  >
    <path
      d="M 26 44 l 0 -20 q 0 -6 4 -6 l 4 0 q 4 0 4 4 l 0 6"
      fill="none"
      stroke=""
      stroke-width="4"
      style="transition: all 0.5s; transform-origin: 26px 50%; transform: translateY(-4px) scaleX(-1);"
    />
     
    <rect
      fill=""
      height="24"
      rx="4"
      ry="4"
      stroke="none"
      width="24"
      x="20"
      y="28"
    />
     
    <g
      fill="white"
      stroke="none"
    >
      <circle
        cx="32"
        cy="37"
        r="3"
      />
       
      <rect
        height="12"
        rx="3"
        ry="3"
        width="2"
        x="31"
        y="34"
      />
    </g>
  </g>
</svg>
`;

exports[`LockIcon locked:true 1`] = `
<svg
  style="transform: rotate(0); fill: currentColor; stroke: currentColor; user-select: none;"
  viewBox="0 0 64 64"
  xmlns="http://www.w3.org/2000/svg"
  xmlns:xlink="http://www.w3.org/1999/xlink"
>
  <g
    fill="white"
  >
    <circle
      cx="32"
      cy="32"
      r="32"
    />
  </g>
   
  <g
    stroke-linecap="round"
  >
    <path
      d="M 26 44 l 0 -20 q 0 -6 4 -6 l 4 0 q 4 0 4 4 l 0 6"
      fill="none"
      stroke=""
      stroke-width="4"
      style="transition: all 0.5s; transform-origin: 26px 50%;"
    />
     
    <rect
      fill=""
      height="24"
      rx="4"
      ry="4"
      stroke="none"
      width="24"
      x="20"
      y="28"
    />
     
    <g
      fill="white"
      stroke="none"
    >
      <circle
        cx="32"
        cy="37"
        r="3"
      />
       
      <rect
        height="12"
        rx="3"
        ry="3"
        width="2"
        x="31"
        y="34"
      />
    </g>
  </g>
</svg>
`;
